### PR TITLE
Add cluster api operator CI jobs

### PR DESCRIPTION
**Pull request by alexander-demichev**
Add basic cluster api operator CI jobs: build, test, verify, apidiff. They are inherited from the core cluster-api repository.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+periodics:
+- name: periodic-cluster-api-operator-test-main
+  interval: 1h
```
Comment (JoelSpeed) [inline config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml:3]: Do we really need a periodic with this frequency? Are we observing the results in testgrid at all? I would expect we probably only need this every 24h if at all

File: config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    base_ref: main
+    path_alias: sigs.k8s.io/cluster-api-operator
+  spec:
+    containers:
+    - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20211217-ea95cec1d4-1.23
+      command:
+      - "./scripts/ci-test.sh"
+      resources:
+        requests:
+          cpu: 7300m
```
Comment (JoelSpeed) [inline config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml:19]: Where did this value come from? Seems quite high

Comment (alexander-demichev) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml:19]: I took it from core capi job config

Comment (JoelSpeed) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml:19]: We can probably use the default unless we know we need this much for running specific tests, might be worth doing a git blame on the capi config and seeing if there's any reasoning for this specific value

Comment (alexander-demichev) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-periodics-main.yaml:19]: https://github.com/kubernetes/test-infra/pull/18439, we can drop the request right now and see if the default works

**Comment by JoelSpeed** [comment]
Changes look good to my knowledge but there's something wrong with the unit tests which looks like it's caused by this change, holding labelling until that has been investigated

**Pull request merged**